dask/highlevelgraph.py

```diff
@@ -119,6 +174,8 @@ def from_collections(cls, name, layer, dependencies=()):
         ...     graph = HighLevelGraph.from_collections(name, layer, dependencies=[self])
         ...     return new_collection(name, graph)
         """
+        if len(dependencies) == 1:
```

Comment:
@jcrist would you prefer that I explicitly change things to use `from_collection` rather than `from_collection`?

Reply:
Since this dispatch happens so quickly, I don't think there'd be any performance improvements from having code directly call `_from_collection`.